src/CMakeLists.txt

```diff
@@ -246,17 +230,151 @@ ADD_DEFINITIONS(-DEIGEN_MPL2_ONLY)
+  FIND_LIBRARY(CLP_LIBRARY NAMES Clp)
+  FIND_LIBRARY(CLPSOLVER_LIBRARY NAMES ClpSolver)
+  FIND_LIBRARY(OSICLP_LIBRARY NAMES OsiClp)
+  IF(NOT DEFINED CLP_LIBRARY AND NOT DEFINED CLP_SOLVER_LIBRARY AND NOT DEFINED OSICLP_LIBRARY)
+    SET(OpenMVG_USE_INTERNAL_CLP ON)
+  ELSE()
+    SET(CLP_LIBRARIES ${CLP_LIBRARY} ${CLPSOLVER_LIBRARY} ${OSICLP_LIBRARY})
+  ENDIF()
+ELSE(CLP_INCLUDE_DIRS)
+  SET(OpenMVG_USE_INTERNAL_CLP ON)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_CLP)
+  SET(CLP_INCLUDE_DIRS
+     ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/osi_clp/Clp/src/
+     ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/osi_clp/Clp/src/OsiClp/)
+  SET(CLP_LIBRARIES lib_clp lib_OsiClpSolver)
+ENDIF()
+
+# ==============================================================================
+# Osi
+# ==============================================================================
+# - internal by default (Osi),
+# - external if OSI_DIR_HINTS and a valid Osi setup is found
+# ==============================================================================
+FIND_PATH(OSI_INCLUDE_DIRS OsiConfig.h ${OSI_DIR_HINTS})
+IF(OSI_INCLUDE_DIRS)
+  FILE(READ "${OSI_INCLUDE_DIRS}/OsiConfig.h" OSI_CONFIG)
+  STRING(REGEX REPLACE ".*OSI_VERSION \"([0-9.]+)\".*" "\\1" OSI_VERSION ${OSI_CONFIG})
+  FIND_LIBRARY(OSI_LIBRARY NAMES Osi)
+  IF(NOT DEFINED OSI_LIBRARY)
+    SET(OpenMVG_USE_INTERNAL_OSI ON)
+  ENDIF()
+ELSE(OSI_INCLUDE_DIRS)
+  SET(OpenMVG_USE_INTERNAL_OSI ON)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_OSI)
+  SET(OSI_INCLUDE_DIRS
+     ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/osi_clp/Osi/src/Osi/)
+  SET(OSI_LIBRARY lib_Osi)
+ENDIF()
+
+# ==============================================================================
+# Lemon
+# ==============================================================================
+# - internal by default (Lemon),
+# - external if LEMON_DIR_HINTS and a valid Lemon setup is found
+# ==============================================================================
+FIND_PATH(LEMON_INCLUDE_DIR config.h ${LEMON_DIR_HINTS})
+IF(LEMON_INCLUDE_DIR)
+  FILE(READ "${LEMON_INCLUDE_DIR}/config.h" LEMON_CONFIG)
+  STRING(REGEX REPLACE ".*LEMON_VERSION \"([0-9.]+)\".*" "\\1" LEMON_VERSION ${LEMON_CONFIG})
+  FIND_LIBRARY(LEMON_LIBRARY NAMES emon)
+  IF(NOT DEFINED LEMON_LIBRARY)
+    SET(OpenMVG_USE_INTERNAL_LEMON ON)
+  ELSE()
+    GET_FILENAME_COMPONENT(LEMON_INCLUDE_DIRS ${LEMON_INCLUDE_DIR} PATH)
+  ENDIF()
+ELSE(LEMON_INCLUDE_DIR)
+  SET(OpenMVG_USE_INTERNAL_LEMON ON)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_LEMON)
+  SET(LEMON_INCLUDE_DIRS
+     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/lemon
+     ${PROJECT_BINARY_DIR}/third_party/lemon)
+  SET(LEMON_LIBRARY lemon)
+ENDIF()
+
+# ==============================================================================
+# Internal CLP/OSI/COINUTILS libraries:
+# ==============================================================================
+ADD_SUBDIRECTORY(dependencies/osi_clp/)
```

Comment:
> Sorry about that, yes please fell free to do a PR for
> https://github.com/openMVG-thirdparty/osi_clp.
> Once it is done, we will be able to update the commit id in the OpenMVG
> repository to make the submodule match the last osi_clp modifications.
>
>
>
I'll do that as solution now, in fact we have this
http://openrisc.rdsor.ro/openMVG-deps.patch . I cannot enforce this upper
level due to local fPIC / STATIC thing.